### DISCUSSION TITLE
🔒 Refund Excess Token (If Any) To Sender in `BatchDistributor`

### DISCUSSION THREAD

**Discussion by jaglinux** [pcaversaccio/snekmate#43] [Ideas]
We need to return excess token in distribute_token(), similar to distribute_ether() which refunds excess ether to msg.sender

If this is ok, I will work on the PR to add the code and test case.

**Comment by pcaversaccio** [comment]
@jaglinux can you please use issues for such requests, thx.

There is no such case as excess tokens for `distribute_token` since this works differently as for `msg.value`. There is no native way to handle tokens in the EVM! First you have to approve the amount you want to transfer and thereafter the exact amount is transferred via `transferFrom` during the `distribute_token` function call. Only if all tokens are distributed, the call will succeed. You don't send tokens directly to the contract but it's a combination of `approve` and `distribute_token`.
